test/common/http/http2/metadata_encoder_decoder_test.cc

```diff
@@ -182,6 +182,20 @@ TEST_F(MetadataEncoderDecoderTest, TestMetadataSizeLimit) {
   cleanUp();
 }
 
+TEST_F(MetadataEncoderDecoderTest, TestTotalPayloadSize) {
+  initialize([](MetadataMapPtr&&) {});
+
+  const std::string payload = std::string(1024, 'a');
+  EXPECT_EQ(0, decoder_->totalPayloadSize());
+  EXPECT_TRUE(
+      decoder_->receiveMetadata(reinterpret_cast<const uint8_t*>(payload.data()), payload.size()));
+  EXPECT_EQ(payload.size(), decoder_->totalPayloadSize());
+  EXPECT_TRUE(
+      decoder_->receiveMetadata(reinterpret_cast<const uint8_t*>(payload.data()), payload.size()));
+  EXPECT_EQ(2 * payload.size(), decoder_->totalPayloadSize());
+  cleanUp();
```

Comment:
This lgtm, but could you add another check to verify that `receiveMetadata` returns false when total payload size exceeds the limit?

Reply:
I'm confused. Isn't this already verified by the preexisting test case MetadataEncoderDecoderTest.TestMetadataSizeLimit, above? Earlier, you didn't want me to duplicate checks that were already being performed in other test cases.